core/bin/block_reverter/src/main.rs

```diff
@@ -106,21 +106,19 @@ async fn main() -> anyhow::Result<()> {
     let opts = Cli::parse();
     let observability_config =
         ObservabilityConfig::from_env().context("ObservabilityConfig::from_env()")?;
-    let log_format: zksync_vlog::LogFormat = observability_config
-        .log_format
-        .parse()
-        .context("Invalid log format")?;
 
-    let mut builder = zksync_vlog::ObservabilityBuilder::new()
-        .with_log_format(log_format)
-        .disable_default_logs(); // It's a CLI application, so we only need to show logs that were actually requested.
-    if let Some(sentry_url) = observability_config.sentry_url {
-        builder = builder
-            .with_sentry_url(&sentry_url)
-            .context("Invalid Sentry URL")?
-            .with_sentry_environment(observability_config.sentry_environment);
-    }
-    let _guard = builder.build();
+    let logs = zksync_vlog::Logs::try_from(observability_config.clone())
+        .context("logs")?
+        .disable_default_logs(); // It's a CLI application, so we only need to show logs that were actually requested.;
+    let sentry: Option<zksync_vlog::Sentry> =
+        TryFrom::try_from(observability_config.clone()).context("sentry")?;
+    let opentelemetry: Option<zksync_vlog::OpenTelemetry> =
+        TryFrom::try_from(observability_config.clone()).context("opentelemetry")?;
+    let _guard = zksync_vlog::ObservabilityBuilder::new()
+        .with_logs(Some(logs))
+        .with_sentry(sentry)
+        .with_opentelemetry(opentelemetry)
+        .build();
```

Comment:
This is only tangentially related, but what is the purpose of us having a guard? I don't see it being used in a meaningful way anywhere and most of the stuff from `ObservabilityBuilder` is being set up as global anyway. As a user I would expect this to also handle `tracing`'s `DefaultGuard` and maybe some other related things.

Maybe it would make sense to rename `build` to `global` or something and not return a guard at all?

Reply:
I see your point, but I feel that the fact that we can't have guards (or other methods of control) for other things here is a bug, not a feature. I don't particularly like the fact that, for example, otlp exporter implicitly installs global handler. It doesn't play well with the node framework either.

A different question is whether we need sentry at all, given that we only use it for panics.

Reply:
I think we could add tracing-related guards here if we really wanted to (open a tracing appender with `WorkerGuard`, then a subscriber with `DefaultGuard`). Not sure about otlp though.

I agree that this is a bug, so my suggestion was more of a "let's not confuse anyone that this behaves a certain way when it really doesn't and we are not investing time in fixing this". In any case not a strong opinion from my side so fine with whatever.